pkg/compactor/bucket_compactor_test.go

```diff
@@ -280,3 +280,71 @@ func TestConvertCompactionResultToForEachJobs(t *testing.T) {
 	require.Equal(t, ulidWithShardIndex{ulid: ulid1, shardIndex: 1}, res[0])
 	require.Equal(t, ulidWithShardIndex{ulid: ulid2, shardIndex: 3}, res[1])
 }
+
+func TestCompactedBlocksTimeRangeVerification(t *testing.T) {
+	tests := map[string]struct {
+		minTime        int64
+		maxTime        int64
+		shouldErr      bool
+		expectedErrMsg string
+	}{
+		"should pass with minTime and maxTime matching the input blocks": {
```

Comment:
[nit] To keep naming consistent, please rename "input blocks" to "source blocks" and "output blocks" to "compacted blocks".

_Comment applies to the rest of the test._